### ISSUE TITLE
Microphone redirection broke on discord --> There must be some incompatible APIs

### ISSUE THREAD

**Issue by XP-Fan** [MoonchildProductions/UXP#1549]
Well I have posted this into the basilisk issue tracker about a week ago. Now that I understood that UXP is the rendering-engine of basilisk I guess it belongs to here. If not, apologies for bothering. Anyway, here is my original post from back then:
"Hey guys, I do know this is not a discord-support but rather about the Basilisk webbrowser. However, this issue seems to be particularly specific to discord (http://discordapp.com) in combination with basilisk. Since about 2 days, the voicechat-functionality is broken, which to me definitely means that there must be some fancy modern API basilisk does not support. I suppose this shall be fixed asap, as it is probably not discord only, they might've been the ones to start, but more pages might decide to use that API/whatever missing thing in the future, so I really really hope Basilisk'll catch up soon... Anyway by this chance I wanna thank you guys for maintaining this as it's the most awesome browser in the world..."

And here is the link to it: https://github.com/MoonchildProductions/Basilisk/issues/15
I Hope this does now reach the right people, if not, I'd be grateful if you passed it on...
Best regards!

**Comment by wolfbeast** [comment]
Most likely duplicate of #1539

**Comment by XP-Fan** [comment]
I don't think so because it broke on a version where it was previously working from one day to another, so I suspect that Discord did some changes on their side. Still it would then mean that there is some required API Basilisk doesn't support. I didn't update before, just after I realized that it is no longer working I switched to the latest version, which did however not change anything. As of now when I try to voicecall someone in Discord, the call button does nothing, the mute/unmute toggle throws me a "browser not supported" dialog. However I'm not an expert on this, I hardly know anything about browsers, but I'd be incredibly happy if you fixed that, especially since I'm blind and Basilisk is - just unlike todays major browsers but like firefox was back in the days - perfectly accessible…

**Comment by hawkeye116477** [comment]
That happens also on Waterfox Classic. User-agent override for `discord.com` fixes this. Seems that they changed domain from `discordapp.com` to `discord.com`.

**Comment by wolfbeast** [comment]
We don't have overrides for discord.com

**Comment by XP-Fan** [comment]
@hawkeye116477: A billion thanks, yes, it was that easy, actually. I added the following line to my user.js:
user_pref("general.useragent.override.discord.com", "Mozilla/5.0 (Windows NT 5.1; rv:68.0) Gecko/20100101 Firefox/68.0");
And this magically restores discord voicechat-compatibility! @Moonchild I strongly suppose that this can now be closed, I'm extremely sorry for bothering though, dunno why I was as stupid as to forget about the domain-change, especially as I regularly read discords changelogs, I might have figured that out on my own… Anyway as my last word I wanna say thank you to all of you guys from the UXP/Basilisk team, as I said it's the most awesome browser, please keep the awesome work!

**Comment by mattatobin** [comment]
You knoe thst doesn't solve anything and it is a temporary workaround at BEST, right?

**Comment by XP-Fan** [comment]
I do not see why? If discord does not know the browser, or they consider it to be an old firefox 52 whose code was never touched ever since 2017 or when it came out, then I guess it's just the only solution to tell them hey, it's not like that. And since the calls work now, it seems like Basilisk actually supports everything which is required. So, I do not see any problem, but maybe you wanna elaborate on that?

**Comment by mattatobin** [comment]
Because Service based WebApps like Discord do not retain compatibility for old versions or non-mainstream versions of browsers for long IF at all.

Spoofing the UA isn't "magic" it is lying to the server and there is no guarantee that the server will respond to the lies forever. THAT is why it is a temporary workaround at best.

You may also want to consider how you phrase things because it could be misinterpreted as you saying that Basilisk is old and insecure.

**Comment by XP-Fan** [comment]
Well, as of now I had the same UA for discordapp.com and it worked like this at least for a year I think. And no, basilisk is neither old nor insecure. It's awesome. What I wanted to say is that the server might consider it to be old, because on its default UA it finds a string Firefox in combination with a 52. Even though there is 68 in between. Well I have no idea how their servers manage those things. It can also be that discord does just not want their calls to run on non-mainstream browsers, but however, it works! And I do not think that it would break anytime soon because I am not spoofing a newer version of Firefox, I just put away the basilisk-prefix…

**Comment by wolfbeast** [comment]
So what exactly did you do to make it work on Basilisk? We can always consider an SSUAO for it.

**Comment by MoonChild** [comment]
You are talking to the wrong MoonChild!

**Comment by wolfbeast** [comment]
@XP-Fan Please be careful who you at-mention. Might be my name but not my username here ;)
MoonChild/Jason sorry for the noise, please just unsubscribe from this issue.

**Comment by XP-Fan** [comment]
I am sorry, but here in the e-mail reply it definitely writes "Moonchild" within the sender.

I'm confused now. Anyway when I just start my basilisk and I visit for example https://www.whatismybrowser.com/detect/what-is-my-user-agent

It reads:

Mozilla/5.0 (Windows NT 5.1; rv:68.9) Gecko/20100101 Goanna/4.6 Firefox/68.9 Basilisk/52.9.0

This way, discord did not work. So, back in 2018, I created an user.js-file and put the following into it:

user_pref("general.useragent.override.discordapp.com", "Mozilla/5.0 (Windows NT 5.1; rv:68.0) Gecko/20100101 Firefox/68.0");

Ever since, discord worked just as expected, till somewhat 2 weeks ago or so. At that point it broke, until I exchanged the domains as hawkeye116477 (I hope this is the right username) suggested. So now, the exact string in my user.js reads:

user_pref("general.useragent.override.discord.com", "Mozilla/5.0 (Windows NT 5.1; rv:68.0) Gecko/20100101 Firefox/68.0");

And now discord-voicecalling works. I'mm just wondering why I put it to 68.0 instead of 68.9… I might try with 68.9 later, but I guess it wouldn't make a difference…

**Comment by mattatobin** [comment]
Your ignorance is astounding. Please take further discussion to the forum.

**Comment by MoonChild** [comment]
I'm literally not the MoonChild you are looking for, good day!
